components/DataTable/src/DataTable/DataTable.cs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace CommunityToolkit.WinUI.Controls;
+
+/// <summary>
+/// A <see cref="DataTable"/> is a <see cref="Panel"/> which lays out <see cref="DataColumn"/>s based on
+/// their configured properties (akin to <see cref="ColumnDefinition"/>); similar to a <see cref="Grid"/> with a single row.
+/// </summary>
+public partial class DataTable : Panel
+{
+    // TODO: We should cache this result and update if column properties change
+    internal bool IsAnyColumnAuto => Children.Any(static e => e is DataColumn { DesiredWidth.GridUnitType: GridUnitType.Auto });
+
+    // TODO: Check with Sergio if there's a better structure here, as I don't need a Dictionary like ConditionalWeakTable
+    internal HashSet<DataRow> Rows { get; private set; } = new();
+
+    internal void ColumnResized()
+    {
+        InvalidateArrange();
+
+        foreach (var row in Rows)
+        {
+            row.InvalidateArrange();
+        }
+    }
```

Comment:
@Sergio0694 wanted to check in with you here, as my instinct was a `ConditionalWeakTable` for safety of reference holding. Though I have the `DataRow` class remove itself on `Unloaded` (though we know how that is in the XAML lifecycle). Right now, I was just trying to get the layout logic to work, and I think that's in a good place now.

Theoretically these should all be owned by the same parent, so not sure if that would cause problems with reference counts. Realizing though that I should probably have the DataRow grab a `WeakReference` to the parent too...

What are your thoughts on this connection? Thanks!

Reply:
I think I'm missing some context on who's keeping alive what. As in, if this is just some internal data structure for a control, I'm not necessarily seeing an issue with it tracking other things and objects it needs. Even if there's some reference cycle, the GC is completely able to handle that. Is there some global (static) state being involved here that's worrying you? Or, could you explain in simple terms what the relationship is between the various components here so I can better understand the scenario and what you're thinking might cause problems? 🙂